bpf/bpf_lxc.c

```diff
@@ -929,18 +928,18 @@ static inline int __inline__ ipv4_policy(struct __sk_buff *skb, int ifindex, __u
 				   REV_NAT_F_TUPLE_SADDR);
 		if (IS_ERR(ret2))
 			return ret2;
-
 	}
 
-	/* Policy lookup is done on every packet to account for packets that
-	 * passed through the allowed consumer. */
 	verdict = policy_can_access(&POLICY_MAP, skb, src_label, tuple.dport,
 				    tuple.nexthdr, sizeof(tuple.saddr),
 				    &tuple.saddr);
-	if (unlikely(ret == CT_NEW)) {
-		if (verdict != TC_ACT_OK)
-			return DROP_POLICY;
 
+	/* Reply packets and related packets are allowed, all others must be
+	 * permitted by policy */
+	if (ret != CT_REPLY && ret != CT_RELATED && verdict != TC_ACT_OK)
+		return DROP_POLICY;
+
+	if (ret == CT_NEW) {
```

Comment:
Same comment here as above seems we can optimize out the policy_can_access for RELATED flows.